lte/gateway/c/session_manager/PipelinedClient.cpp

```diff
@@ -249,18 +324,18 @@ void AsyncPipelinedClient::set_upf_session(
 
 void AsyncPipelinedClient::deactivate_flows_for_rules_for_termination(
     const std::string& imsi, const std::string& ip_addr,
-    const std::string& ipv6_addr, const Teids teids,
+    const std::string& ipv6_addr, const std::vector<Teids>& teids,
     const RequestOriginType_OriginType origin_type) {
-  MLOG(MDEBUG) << "Deactivating all rules and default drop flows "
-               << "for " << imsi << ", ipv4: " << ip_addr
-               << ", ipv6: " << ipv6_addr << ", agw teid: " << teids.agw_teid()
-               << ", enb teid: " << teids.enb_teid()
-               << ", origin_type: " << request_origin_type_to_str(origin_type);
-
-  RulesToProcess empty_to_process = std::vector<RuleToProcess>{};
-  auto req                        = create_deactivate_req(
-      imsi, ip_addr, ipv6_addr, teids, empty_to_process, origin_type, true);
-  deactivate_flows(req);
+  for (const Teids& t : teids) {
+    MLOG(MDEBUG) << "Deactivating all rules and default drop flows "
+                 << "for " << imsi << ", ipv4: " << ip_addr
+                 << ", ipv6: " << ipv6_addr << ", agw teid: " << t.agw_teid()
+                 << ", enb teid: " << t.enb_teid() << ", origin_type: "
+                 << request_origin_type_to_str(origin_type);
+    DeactivateFlowsRequest req =
+        make_deactivate_req(imsi, ip_addr, ipv6_addr, t, origin_type, true);
+    deactivate_flows(req);
+  }
```

Comment:
why don't pipelined do the same as we do with the case of empty rules? If teid is empty, then deactivate all for that IMSI

Reply:
This doesn't work because there could be multiple sessions for that IMSI. :/ We can't accidentally remove flows for a separate session for the same UE. (CC @koolzz)